src/StreamingDataFrames/streamingdataframes/models/topics.py

```diff
@@ -23,6 +24,20 @@
 
 logger = logging.getLogger(__name__)
 
+__all__ = (
+    "TopicCreationConfigs",
+    "Topic",
+)
+
+
+class TopicCreationConfigs:
+    def __init__(
+        self, num_partitions: int, replication_factor: int, optionals: dict = None
```

Comment:
Would it be cleaner if we had a dataclass here instead, and defined all the default values on the class level?

`retention.bytes` and `retention.ms` could also be converted to `retention_bytes` and `retention_minutes` and passed to `post_topic()` as they are.

And we could also remove the default values from [here](https://github.com/quixio/quix-streams/pull/202/files#diff-49b0ec4d7de4eb00fd3a27a08e2edc351bfea1468f4ce4455b88ce573c1a0e5eR136-R139) and from `post_topic()` so they're not spread across the code

Reply:
Yeah, I think we can use a dataclass! 

The main reason I didn't include `retention.bytes` and `retention.minutes` as class values (written as you suggested with underscores) is that they are technically optional wrt topic creation, so I only made the required arguments as attributes. 

If we think we'll only use this with Quix ever, then I'm fine with changing that...or overall we surface the most common ones and leave some of the more obscure ones under "optional" (which I'd def say those two are common enough). I can see the argument either way.

Similarly, It's definitely awkward to have the default spread across the code, but wasn't sure the best way to do it since you could create topics with the API or with the Quix library, and not sure if we should have defaults in general for partition count and rep factor, and whether it should be set to the Quix defaults vs the Kafka defaults (aka use it for non-quix stuff)?

TL;DR...do we expect to use this beyond Quix stuff and/or should we be opinionated about the "common" topic options and hide the rest?

Reply:
Well, this is the code to work with Quix platform API, so I'd consider this as Quix-only. 
This object is only used to specify parameters for the API call.

Reply:
I suggest we also move it to `platforms/quix` to avoid confusion